RoddyCore/src/de/dkfz/roddy/knowledge/nativeworkflows/GenericJobInfo.groovy

```diff
@@ -40,18 +40,19 @@ class GenericJobInfo {
     final String otherSettings
     final String inlineScript
 
-    GenericJobInfo(ExecutionContext context, BEGenJI jInfo, String inlineScript = null) {
+    GenericJobInfo(ExecutionContext context, BEGenJI jInfo, String inlineScript = null, BufferUnit bufferUnit = BufferUnit.G) {
         executionContext = context
         jobName = jInfo.jobName
         id = jInfo.id
         parameters = jInfo.parameters
         parentJobIDs = jInfo.parentJobIDs
-        walltime = jInfo.walltime
-        memory = jInfo.maxMemory
-        memoryBufferUnit = jInfo.memoryBufferUnit
-        cpus = jInfo.maxCpus
-        nodes = jInfo.maxNodes
-        queue = jInfo.queue
+        walltime = jInfo.usedResources.walltime
+        assert(jInfo.usedResources.mem.toLong(bufferUnit) <= Integer.MAX_VALUE)
```

Comment:
Integer or Long max value?

Reply:
Well, the field is integer. Now it is at least checked that the conversion is o.k. We could make the field Long, though, if it makes sense. It really depends on the BufferUnit. For G, Integer should probably be fine for some years :-)